pkg/apis/apps/v1alpha1/servicebinding_types.go

```diff
@@ -0,0 +1,108 @@
+package v1alpha1
+
+import (
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+
+// ServiceBindingSpec defines the desired state of ServiceBindingSpec
+type ServiceBindingSpec struct {
+	// MountPathPrefix is the prefix for volume mount
+	// +optional
+	MountPathPrefix string `json:"mountPathPrefix,omitempty"`
+
+	// EnvVarPrefix is the prefix for environment variables
+	// +optional
+	EnvVarPrefix string `json:"envVarPrefix,omitempty"`
+
+	// DataMapping is used to map CR/CSV/CRD attributes to Custom env variables
+	// +optional
+	DataMapping []corev1.EnvVar `json:"dataMapping,omitempty"`
+
+	// Services is used to identify multiple backing services.
+	Services []Service `json:"services"`
+
+	// Application is used to identify the application connecting to the
+	// backing service operator. When not provided, a managed binding secret
+	// is created but not bound to any application workload.
+	// +optional
+	Application Application `json:"application,omitempty"`
+
+	// DetectBindingResources is flag used to bind all non-bindable variables from
+	// different subresources owned by backing operator CR.
+	// +optional
+	DetectBindingResources bool `json:"detectBindingResources,omitempty"`
+}
+
+// ServiceBindingStatus defines the observed state of ServiceBinding
+// +k8s:openapi-gen=true
+type ServiceBindingStatus struct {
+	// Conditions describes the state of the operator's reconciliation functionality.
+	Conditions []conditionsv1.Condition `json:"conditions"`
+	// SecretRef is the name of the intermediate binding secret generated by
+	// the controller.
+	Secret corev1.LocalObjectReference `json:"secretRef"`
+	// ApplicationObjects contains all the application objects filtered by label
+	// +optional
+	Applications []BoundApplication `json:"applications,omitempty"`
```

Comment:
Since we cannot bind to more than one application, we should not have a list here.

Reply:
@DhritiShikhar it is possible to have more than one application even if you can only declare one entry in `spec.application`.


`Application` embeds `metav1.LabelSelector` as follows:

```go
// Application defines the selector based on labels and GVR
type Application struct {
	// +optional
	LabelSelector               *metav1.LabelSelector `json:"labelSelector,omitempty"`
	metav1.GroupVersionResource `json:",inline"`
	corev1.LocalObjectReference `json:",inline,omitempty"`
}
```

It is then used in [`Binder.search()`](https://github.com/sbose78/service-binding-operator/blob/b4dc1abda582e2fe27d93527042738efdc764031/pkg/controller/servicebinding/binder.go#L90-L93) to list all resources matching the given GVR in the given namespace:

```go
	objects, err := b.dynClient.Resource(gvr).Namespace(ns).List(opts)
	if err != nil {
		return nil, ApplicationNotFound
        }
```

This call to `List()` can return multiple resources if multiple resources exist with labels declared in `spec.application.matchLabels`.